tests/ut/test_core_setup.py

```diff
@@ -0,0 +1,265 @@
+        self._init_global_admin()
+        self._global_admin.get = mock.MagicMock(side_effect=[
+            TestCoreSetup._generate_status_response(SetupWizardStage.Server, SetupWizardStatus.NA, ''),
+            TestCoreSetup._generate_status_response(SetupWizardStage.Portal, SetupWizardStatus.NA, ''),
+            TestCoreSetup._generate_status_response(SetupWizardStage.Finish, SetupWizardStatus.NA, '')
+        ])
+        mock_startup_wait = self.patch_call("cterasdk.core.startup.Startup.wait")
+
+        setup.Setup(self._global_admin).init_master(self._admin_username, self._admin_email,
+                                                    self._admin_first_name, self._admin_last_name, self._admin_password, self._domain)
+
+        setup_status_url = '/%s/setup/status' % (self._global_admin.context)
+        self._global_admin.get.assert_has_calls(
+            [
+                mock.call(setup_status_url, use_file_url=True),
+                mock.call(setup_status_url, use_file_url=True),
+                mock.call(setup_status_url, use_file_url=True)
+            ]
+        )
+        self._global_admin.execute.assert_called_once_with('/%s/public' % (self._global_admin.context),
+                                                           'init', mock.ANY, use_file_url=True)
+        expected_param = self._get_init_portal_param()
+        actual_param = self._global_admin.execute.call_args[0][2]
+        self._assert_equal_objects(actual_param, expected_param)
+
+        params = TestCoreSetup._get_init_server_params(ServerMode.Master)
+        form_data = TestCoreSetup._get_form_data(params)
+        self._global_admin.multipart.assert_called_once_with('/%s/setup' % (self._global_admin.context), form_data, use_file_url=True)
+
+        mock_startup_wait.assert_called_once()
+
+    def test_init_master_already_finished(self):
+        self._init_global_admin()
+        get_function = mock.MagicMock(side_effect=[
+            TestCoreSetup._generate_status_response(SetupWizardStage.Finish, SetupWizardStatus.NA, '')
+        ])
+        self._global_admin.get = get_function
+        mock_startup_wait = self.patch_call("cterasdk.core.startup.Startup.wait")
+
+        setup.Setup(self._global_admin).init_master(self._admin_username, self._admin_email,
+                                                    self._admin_first_name, self._admin_last_name, self._admin_password, self._domain)
+
+        mock_startup_wait.assert_called_once()
+
+    def test_init_application_server_success_password(self):
+        self._test_init_application_server_success(SlaveAuthenticaionMethod.Password)
+
+    def test_init_application_server_success_pk(self):
+        self._test_init_application_server_success(SlaveAuthenticaionMethod.PrivateKey)
+
+    def _test_init_application_server_success(self, authentication_method):
+        self.patch_call("time.sleep")
+        self._init_global_admin()
+        get_function = mock.MagicMock(side_effect=[
```

Comment:
Still, no need for the get_function and execute_function variables

Reply:
Done